dcos/subcommand.py

```diff
@@ -324,6 +408,99 @@ def _find_virtualenv(bin_directory):
     return virtualenv_path
 
 
+def _copy_and_cleanup(src, dst):
+    """Copy file or dst src to the file or directory dst and remove dst if failure
+
+    :param src: source file
+    :type src: str
+    :param dst: destination file or directory
+    :type dst: str
+    :returns: None
+    """
+
+    try:
+        if os.path.isfile(src):
+            util.sh_copy(src, dst)
+        else:
+            shutil.copytree(src, dst)
```

Comment:
We should use `shutil.move`. It will figure out if a filesystem move is possible and use that else it will perform a copy and remove. https://docs.python.org/2/library/shutil.html#shutil.move